coins/monero/src/wallet/send/mod.rs

```diff
@@ -750,7 +755,8 @@ impl Eventuality {
         view_tag: Some(expected.view_tag).filter(|_| matches!(self.protocol, Protocol::v16)),
       } != actual) ||
         (Some(&expected.commitment.calculate()) != tx.rct_signatures.base.commitments.get(o)) ||
-        (Some(&expected.amount) != tx.rct_signatures.base.ecdh_info.get(o))
+        (Some(&EcdhInfo::Bulletproof { amount: expected.amount }) !=
```

Comment:
I think this is safe, yet I need to check/potentially update documentation.

Reply:
Eventuality should likely embed an RctType into itself...

Reply:
We do embed Protocol making this safe so long as Eventuality bounds the protocol.